Dashboards: Fix missing Ctrl+O keyboard shortcut for crosshair toggle

ivanortegaalba opened the pull request:
> Restores missing Ctrl+O/Cmd+O keyboard shortcut functionality in dashboard scenes

## Problem

The Ctrl+O keyboard shortcut for toggling shared crosshairs was missing in scenes-based dashboards, causing the browser's file open dialog to appear instead of cycling through crosshair modes. This functionality worked in legacy dashboards but was not implemented in the scenes architecture.

## Solution

- Added missing mod+o keybind to dashboard scenes keyboard shortcuts
- Implemented crosshair state cycling: Default -> Shared crosshair -> Shared tooltip -> Default  
- Added comprehensive unit tests (21 tests) for keyboard shortcuts functionality
- Added e2e test to verify shortcut works and prevents browser file dialog
- Ensured parity between legacy and scenes dashboard implementations

## How to test

1. Create or open a dashboard with scenes enabled
2. Press Ctrl+O (or Cmd+O on Mac)
3. Verify it cycles through crosshair modes instead of opening browser file dialog
4. Open dashboard settings to confirm the Graph tooltip setting changes
5. Run unit tests: `npm test -- --testPathPattern="keyboardShortcuts.test.ts"`
6. Run e2e test: `npm run e2e:playwright -- --grep "ctrl\+o should toggle shared crosshair"`

ivanortegaalba committed:
Dashboard Scenes: Fix missing Ctrl+O keyboard shortcut for crosshair toggle

- Add missing mod+o keybind to dashboard scenes keyboard shortcuts
- Implement crosshair state cycling (Default -> Crosshair -> Tooltip -> Default)
- Add comprehensive unit tests for keyboard shortcuts functionality
- Add e2e test to verify shortcut works and prevents browser file dialog
- Fix ensures parity between legacy and scenes dashboard implementations

Fixes issue where Ctrl+O/Cmd+O was opening browser file dialog instead of
toggling shared crosshair modes in scenes-based dashboards.

ashharrison90 commented on e2e-playwright/various-suite/keybinds.spec.ts:
you know what i'm gonna say here... 😅 

pls don't `await page.waitForTimeout` anywhere 🙏

ivanortegaalba replied:
Removed ❤️

ashharrison90 reviewed (commented):
implementation looks great! 🙌

ivanortegaalba committed:
Remove waitForTimeout from e2e test

- Replace arbitrary timeouts with proper element waiting
- Use waitFor with visible state instead of setTimeout
- Improve test reliability and follow Playwright best practices

dprokop reviewed (approved):
overall LGTM, but i don't understand why we need to trigger `LegacyGraphHoverClearEvent`. 


Ref https://github.com/grafana/support-escalations/issues/18417

ivanortegaalba commented on public/app/features/dashboard-scene/scene/keyboardShortcuts.ts:
```diff
@@ -168,6 +168,20 @@ export function setupKeyboardShortcuts(scene: DashboardScene) {
+        const currentSync = cursorSync.state.sync;
+        const nextSync = (currentSync + 1) % 3;
+        cursorSync.setState({ sync: nextSync });
+        appEvents.publish(new LegacyGraphHoverClearEvent());
```
It is exported in `@grafana/data` and emitted in the old arch. So if we don't want to introduce a breaking change, we should do it. 3rd-party panel plugins could rely on those to change their pointer configuration.

ivanortegaalba committed:
Optimize e2e test for crosshair keyboard shortcut

- Remove unnecessary console logging and timeout settings
- Simplify assertions to only check the currently selected radio button
- Improve test performance by reducing DOM queries
- Focus on essential functionality verification

ivanortegaalba committed:
Fix linting
